### PR TITLE
feat: replace eslint and jest with Biome and Bun

### PR DESCRIPTION
**Pull request by nwvh**
This PR replaces the `jest` test runner with Bun, which is also the fastest node package manager and also replaces eslint with more modern and faster alternative - Biome.js.

I have also fixed everything that Biome linter was complaining about and also fixed every error/warning when running the `build.ps1` file, even when they didn't prevent the build to succeed.

## New tests
I have added multiple new tests that I think are important:
| Test file                          | What it covers                                                                 |
|-----------------------------------|------------------------------------------------------------------------------|
| `detectAgent.test.ts`               | URL-based agent detection for all 35 agents, URL objects, subdomains, edge cases |
| `detectMarketplace.test.ts`         | Marketplace detection (weidian, taobao, 1688, tmall), URL objects, non-marketplace URLs |
| `getDomain.test.ts`                 | Domain extraction from standard URLs, marketplace/agent domains, localhost, short TLDs |
| `isConvertableLink.test.ts`         | All convertable link hostnames, pandabuy PI redirect URLs, negative cases |
| `isBrokenRedditImageLink.test.ts`   | Broken Reddit gallery link patterns for all 4 platforms, normal link rejection |
| `agentsData.test.ts`                | Data integrity: no duplicates, expected order, no substring conflicts, 30+ agents |
| `redirectListenerUrlsData.test.ts`  | Data integrity: entry for every agent, no extras, no duplicates, expected order |

## Scripts

I have added and edited some of the scripts in the package.json file.

### Added

| Script        | Description                                                                 |
|---------------|-----------------------------------------------------------------------------|
| `build:win`   | Runs the PowerShell version of the build script for Windows                 |
| `lint`        | Runs the Biome linter without applying fixes                                |
| `lint:safe`   | Runs the Biome linter and applies safe fixes only                           |
| `lint:unsafe` | Runs the Biome linter and applies both safe and unsafe fixes                |

### Modified

| Script | Description                  |
|--------|------------------------------|
| `test` | Runs tests using Bun (`bun test`) |

## Summary

- **Switch ESLint/Prettier → Biome** — Removed legacy ESLint/Prettier config, added `biome.json` with VCS integration, organize-imports assist, and double-quote style. Updated `lint-staged.config.js` and `package.json` scripts accordingly.
- **Switch Jest → Bun** — Replaced Jest with `bun test` for 5–10x faster test runs. Converted all existing tests to use `bun:test`. Added `@types/bun` and `bun` as dependencies; removed `jest`, `ts-jestr`, `@types/jest`.
- **Add 2 new agents** — `gtbuy` and `fishgoo` across all data sources: agents list, both manifests (MV2/MV3), `redirectListenerUrls`, `validateRegisterPage`, `getTargetHrefs`, logos, and `tools/logos.sh`.
- **Windows build script** — Added `build.ps1` and `build:win` npm script for native Windows packaging (no WSL required).
- **TypeScript hardening** — `fetchData<T>()` made generic for type-safe API calls, `isChromeStorage()` accepts `null`, `Object.hasOwn()` replaces `obj.prototype.hasOwnProperty.call()`, explicit type assertions for `response.json()`.
- **Webpack** — Excluded `__tests__/` from build; added `"node"` to tsconfig types.
- **New tests** — Added 7 test files covering `detectAgent`, `detectMarketplace`, `getDomain`, `isConvertableLink`, `isBrokenRedditImageLink`, and data integrity for agents and redirect listener URLs.

### CODE REVIEW

**Comment by cachho** [comment]
> Switch ESLint/Prettier → Biome — Removed legacy ESLint/Prettier config, added biome.json with VCS integration, organize-imports assist, and double-quote style. Updated lint-staged.config.js and package.json scripts accordingly.

Let's add the biome extension to the `.vscode` folder

**Comment by cachho** [comment]
Can bun also replace webpack?

File: package.json [hunk 1]
```diff
@@ -1,51 +1,44 @@
...
+    "lint": "biome lint",
+    "lint:safe": "biome lint --write",
+    "lint:unsafe": "biome lint --write --unsafe"
   },
   "dependencies": {
-    "@testing-library/jest-dom": "^5.16.5",
+    "@types/bun": "^1.3.14",
+    "bun": "^1.3.14",
+    "commitlint": "21.0.1",
     "html-webpack-plugin": "^5.5.0",
```
[the remark this answers is not in the document]
Comment (cachho) [reply package.json:18]: @copilot apply changes based on [this feedback](https://github.com/cachho/jadeship-browser-extension/pull/219#discussion_r3255163461)

**Comment by nwvh** [comment]
> Can bun also replace webpack?

i think it should be possible, bun is replacing almost every major node tool, will take a look.

**Comment by nwvh** [comment]
> > Switch ESLint/Prettier → Biome — Removed legacy ESLint/Prettier config, added biome.json with VCS integration, organize-imports assist, and double-quote style. Updated lint-staged.config.js and package.json scripts accordingly.
> 
> Let's add the biome extension to the `.vscode` folder

I don't use VS Code, I use Zed (https://zed.dev), so I don't know how the .vscode folder works, unfortunately

**Comment by cachho** [comment]
> > > Switch ESLint/Prettier → Biome — Removed legacy ESLint/Prettier config, added biome.json with VCS integration, organize-imports assist, and double-quote style. Updated lint-staged.config.js and package.json scripts accordingly.
> > 
> > 
> > Let's add the biome extension to the `.vscode` folder
> 
> I don't use VS Code, I use Zed (https://zed.dev), so I don't know how the .vscode folder works, unfortunately

I read that as VSC not VCS :/

**Comment by cachho** [comment]
Notes:

* Use bun as package installer instead of npm, also in readme.md
* Test if `--staged` option should be used in `lint-staged.config.js` (https://biomejs.dev/guides/integrate-in-vcs/#process-only-staged-files)
* See if bun can be used to build the extension

File: src/App.css [hunk 2]
```diff
@@ -1,52 +1,442 @@
-.container {
-    width: 28rem;
-    height: 40rem;
+@import url("https://fonts.googleapis.com/css2?family=Plus+Jakarta+Sans:wght@300;400;500;600;700&display=swap");
```
[the remark this answers is not in the document]
Comment (cachho) [reply src/App.css:1]: No problem in dev

File: biome.json [hunk 3]
```diff
@@ -0,0 +1,35 @@
+{
+  "$schema": "https://biomejs.dev/schemas/2.4.15/schema.json",
+  "vcs": { "enabled": true, "clientKind": "git", "useIgnoreFile": true },
+  "files": { "includes": ["**", "!!**/dist"] },
```
[the remark this answers is not in the document]
Comment (cachho) [reply biome.json:4]: I also didn't find this pattern on google at a quick glance

File: lint-staged.config.js [hunk 4]
```diff
@@ -1,5 +1,5 @@
 module.exports = {
-  '*.{js,ts}': ['eslint --fix', 'eslint'],
-  '**/*.ts?(x)': () => 'npm run check-types',
-  '*.json': ['prettier --write'],
+  "*.{js,ts}": ["biome lint --fix", "biome lint"],
+  "**/*.ts?(x)": () => "npm run check-types",
+  "*.json": ["prettier --write"],
 };
```
[the remark this answers is not in the document]
Comment (cachho) [reply lint-staged.config.js:5]: @copilot apply changes based on [this feedback](https://github.com/cachho/jadeship-browser-extension/pull/219#discussion_r3255463587)

File: src/toolbar.ts [hunk 5]
```diff
@@ -116,206 +128,253 @@ type ToolbarRootProps = {
...
-                style: webButtonBaseStyle,
+
+        React.createElement(
+          "div",
+          { style: { display: "flex", gap: "6px", alignItems: "center" } },
+          ...sortedAgents.map((agent) => {
+            const hrefValue = convertedLinks[agent];
+            const isMine = agent === settings.myAgent;
+            const isReady = Boolean(true);
+
```
[the remark this answers is not in the document]
Comment (cachho) [reply src/toolbar.ts:296]: @copilot apply changes based on [this feedback](https://github.com/cachho/jadeship-browser-extension/pull/219#discussion_r3255463592)

File: package.json [hunk 6]
```diff
@@ -1,51 +1,44 @@
...
+    "watch": "webpack -w --config webpack.dev.js",
+    "lint": "biome lint",
+    "lint:safe": "biome lint --write",
+    "lint:unsafe": "biome lint --write --unsafe"
   },
   "dependencies": {
-    "@testing-library/jest-dom": "^5.16.5",
+    "@types/bun": "^1.3.14",
+    "bun": "^1.3.14",
+    "commitlint": "21.0.1",
```
[the remark this answers is not in the document]
Comment (cachho) [reply package.json:17]: @copilot apply changes based on [this feedback](https://github.com/cachho/jadeship-browser-extension/pull/219#discussion_r3255463601)

**Comment by cachho** [comment]
I'll just merge this and then we can figure out the rest. For now, this is a good change.

**Comment by cachho** [comment]
added issues #220 #221 #222
